client/modules/wj-util/src/decorators.ts

```diff
@@ -0,0 +1,39 @@
+import { perfy } from "./index"
+
+/**
+ * Decorator for measuring the performance of a function or method.
+ * @example
+ * ```
+ * let perf = 0
+ * class foo {
+ *   ;@measure(time => perf = time)
+ *   method() { return "some-expensive-calculation" }
+ * }
+ * ```
+ * @param callback - Callback to fire with the performance measurement taken.
+ */
+export function measure(callback: (perf: number) => void) {
+  // eslint-disable-next-line @typescript-eslint/ban-types
+  return (_target: Object, _propertyKey: string, descriptor: PropertyDescriptor) => {
+    const method = descriptor.value
+    const async = method.constructor.name === "AsyncFunction"
+
+    if (!async) {
```

Comment:
If there is both an if and else branch, I find having the main condition be positive (i.e. not negated) to be more readable.